src/Chapter2.hs

```diff
@@ -348,9 +360,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
 
+firstHalf :: [a] -> [a]
+firstHalf l = take len l
+  where len = (length l) `div` 2
```

Comment:
Neat usage of `where`! Note that brackets are not needed when you use the function in the infix form 🙂 

```suggestion
  where len = length l `div` 2
```